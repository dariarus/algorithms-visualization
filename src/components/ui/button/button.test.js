import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';

import { Button } from './button';

describe('Button Component', () => {
  it('Button has text', () => {
    const tree = renderer
      .create(<Button text="Кнопка с текстом" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Button has no text', () => {
    const tree = renderer
      .create(<Button />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Button is disabled', () => {
    const tree = renderer
      .create(<Button disabled={true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Button has loader', () => {
    const tree = renderer
      .create(<Button isLoader={true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Button click is working correctly', () => {
    window.alert = jest.fn();

    // Рендерим компонент
    render(<Button text="кнопка" onClick={() => {
      alert('Button click works!')
    }
    }/>)

    // Находим элемент кнопки
    const button = screen.getByText("кнопка")

    // Имитируем нажатие на кнопку
    fireEvent.click(button);

    // Проверяем, что alert сработал с правильным текстом предупреждения
    expect(window.alert).toHaveBeenCalledWith('Button click works!');
  });
})
