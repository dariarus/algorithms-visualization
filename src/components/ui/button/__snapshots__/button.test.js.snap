// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Button has loader 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button Component Button has no text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button Component Button has text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Кнопка с текстом
  </p>
</button>
`;

exports[`Button Component Button is disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
